### PR TITLE
[AI] test: Re-enable testing in CI

### PR DESCRIPTION
**Pull request by kdestin**
# Description

This pull request makes the necessary code changes to allow us to turn on test runs in CI for the AI service directory.

Specifically, this pull request:

* Fixes some import errors
* Adds stricter bounds for previously unbounded dependencies
  * Our test CI exercises the lower bounds of dependencies.
* Remedies some packaging issues in `setup.py` (see `package_data`)
* **Needs Review**: Switches the `mlflow` dependency to `mlflow-skinny`
* Adds a placeholder test in `azure-ai-resources` to allow pytest to run without returning non-zero.

Note that test runs on Python 3.7 and Python 3.12 were disabled:

* `azure-ai-generative` does not support Python 3.7 (which is already end of life)
* `azure-ai-generative`'s `index` extra depends on `azureml-dataprep[parquet]` which is not installable on Python 3.12. We plan to turn on testing on Python 3.12 once this is resolved.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by kdestin** [comment]
Green test run https://dev.azure.com/azure-sdk/public/_build/results?buildId=3217833&view=results

**Review by YusakuNo1 — APPROVED** [review]

File: eng/pipelines/templates/stages/platform-matrix-ai.json
Comment (needuv) [inline eng/pipelines/templates/stages/platform-matrix-ai.json:1]: what is this file for?

Comment (needuv) [reply eng/pipelines/templates/stages/platform-matrix-ai.json:1]: oh nvm, I see why it's needed in ci.yml. ignore!

File: sdk/ai/azure-ai-generative/setup.py [hunk 1]
```diff
@@ -63,14 +63,21 @@
...
         # NOTE: To avoid breaking changes in a major version bump, all dependencies should pin an upper bound if possible.
-        "azure-ai-resources",
-        "azureml-telemetry",
-        "mlflow<3",
-        "azure-mgmt-authorization",
-        "opencensus-ext-azure<2.0.0",
+        "azure-ai-resources!=0.0.0b0",
+        "azureml-telemetry~=1.0,>=1.43.0",
+        "mlflow-skinny<3",
+        "azure-mgmt-authorization>=2.0.0",
```
Comment (needuv) [inline sdk/ai/azure-ai-generative/setup.py:79]: remove `azure-mgmt-authorization`

File: sdk/ai/azure-ai-resources/setup.py [hunk 2]
```diff
@@ -66,9 +66,9 @@
     python_requires="<4.0,>=3.8",
     install_requires=[
         # NOTE: To avoid breaking changes in a major version bump, all dependencies should pin an upper bound if possible.
-        "azure-ai-ml",
-        "mlflow<3",
-        "azure-mgmt-authorization"
+        "azure-ai-ml~=1.0",
+        "mlflow-skinny<3",
+        "azure-mgmt-authorization>=0.40.0"
```
Comment (needuv) [inline sdk/ai/azure-ai-resources/setup.py:71]: remove `mgmt-authorization` here too

**Comment by needuv** [comment]
follow ups to this PR:

1. can we enable sphinx? The settings are here: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-generative/pyproject.toml#L8C1-L8C22
2. can we enable API stub gen in addition to tests? We should be unblocked now that we can install the full suite of dependencies.

File: sdk/ai/azure-ai-resources/setup.py (hunk 2 above)
Comment (kdestin) [reply sdk/ai/azure-ai-resources/setup.py]: Resolved by https://github.com/Azure/azure-sdk-for-python/pull/32770/commits/3e7739f2afab4ef36e3c915950bf4309676954d9

**Comment by kdestin** [comment]
> follow ups to this PR:
> 
> 1. can we enable sphinx? The settings are here: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-generative/pyproject.toml#L8C1-L8C22

I'll follow up on this is a new PR.


> 2. can we enable API stub gen in addition to tests? We should be unblocked now that we can install the full suite of dependencies.

I can look into this after this PR gets merged.

Though, one note is that were aren't installing _all_ dependencies. The last time I evaluated it, `promptflow` conflicts with `azure-ai-ml`. So currently, the extras that depend on `promptflow` aren't being installed in `dev-requirements.txt`

**Comment by needuv** [comment]
> > follow ups to this PR:
> > 
> > 1. can we enable sphinx? The settings are here: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-generative/pyproject.toml#L8C1-L8C22
> 
> I'll follow up on this is a new PR.
> 
> > 2. can we enable API stub gen in addition to tests? We should be unblocked now that we can install the full suite of dependencies.
> 
> I can look into this after this PR gets merged.
> 
> Though, one note is that were aren't installing _all_ dependencies. The last time I evaluated it, `promptflow` conflicts with `azure-ai-ml`. So currently, the extras that depend on `promptflow` aren't being installed in `dev-requirements.txt`

That's fine. I don't think pf is used as a top-level import, so we should be good there.

**Review by needuv — APPROVED** [review]

**Pull request merged**